lint

```diff
@@ -171,6 +171,27 @@ lint_files() {
     exit $lint_result
 }
 
+filter_out() {
+    local globs="$1"
+    if [ -n "$globs" -a -r "$globs" ]; then
+        # Files to debug the regular expressions, as they get quite complex:
+        local files_in="$(mktemp /tmp/lint_files_in_XXX.txt)"
+        local files_out="$(mktemp /tmp/lint_files_out_XXX.txt)"
+        # Globs blacklist found: filter out stdin before passing it to stdout:
+        while read -r filename; do
+            # Remove comments. Remove empty lines. Expand directories (e.g. "bin/" -> "bin/.*"). And then regexify globs.
+            # These regexes are then passed to grep to filter out matching elements from stdin:
+            echo "${filename}" | tee -a "$files_in" | grep -vEf <(sed '/^#.*$/d ; /^\s*$/d ; s/^\(.*\/\)$/\1*/g ; s/\([.|]\)/\\\1/g ; s/\?/./g ; s/\*/.*/g ; s/\(.*\)/\^\1\$/g' "$globs") | tee -a "$files_out"
+        done
+        # List all files we have just filtered out:
+        [ -n "$DEBUG" ] && echo >&2 "Files not linted:" && grep >&2 -xvFf "$files_out" "$files_in"
+        rm -f "$files_in" "$files_out"
+    else
+        # No globs blacklist was provided: simply propagate stdin to stdout:
+        while read -r filename; do echo "${filename}"; done
```

Comment:
you could replace this line simply with `cat`, though I suppose there's an argument that this is more readable.